gmail_yaml_filters/upload.py

```diff
@@ -275,7 +277,9 @@ def get_gmail_credentials(
         flow = oauth2client.client.flow_from_clientsecrets(client_secret_path, scopes)
         flow.user_agent = application_name
         flags_parser = argparse.ArgumentParser(parents=[oauth2client.tools.argparser])
-        credentials = oauth2client.tools.run_flow(flow, store, flags=flags_parser.parse_args([]))
+        flags=flags_parser.parse_args([])
+        flags.noauth_local_webserver = True
+        credentials = oauth2client.tools.run_flow(flow, store, flags)
```

Comment:
I'm not sure I understand what this fixes, though I believe you that it helped somewhere. Do you have steps to reproduce an error?

Reply:
I was doing this from a completely headless system, so I can't start up chrome or whatever to do all the oath.

This allows all the oath to happen on the command line.